src/components/CalculatorLine.js

```diff
@@ -1,19 +1,68 @@
-import React, {useState, useLayoutEffect} from 'react'
+import React, {useEffect, useState} from 'react'
 
 import {Row, Col, InputGroup, FormControl} from 'react-bootstrap'
 
-function createHandleChange(setFunction) {
-    return e => {
-        setFunction(e.target.value)
+
+function ReadOnlyField({placeholder, label, value, readOnly}) {
+    return (
+        <FormControl
+            placeholder={placeholder}
+            aria-label={label}
+            value={value}
+            readOnly={readOnly}
+        />
+    )
+}
+
+function InputField({placeholder, label, setFunction, type}) {
+    type = type ? type : 'number'
+    return (
+        <FormControl
+            placeholder={placeholder}
+            aria-label={label}
+            min='0'
+            type={type}
+            onChange={e => setFunction(e.target.value)}
+        />
+    )
+}
+
+function MoneyInputGroup(props) {
+    let input
+    if (props.readOnly) {
+        input = (
+            <ReadOnlyField
+                placeholder={props.placeholder}
+                label={props.label}
+                value={props.value}
+                readOnly={props.readOnly}
+            />
+        )
+    } else {
+        input = (
+            <InputField
+                placeholder={props.placeholder}
+                label={props.label}
+                setFunction={props.setFunction}
+            />
+        )
     }
+    return (
+        <InputGroup>
+            <InputGroup.Prepend>
+                <InputGroup.Text>$</InputGroup.Text>
+            </InputGroup.Prepend>
+            {input}
```

Comment:
```suggestion
            {
              props.readOnly ? <ReadOnlyField {...props} /> : <InputField {...props} />
            }
```

If you want to pass the props your component is receiving straight to the component you are rendering (like in this case), you can use the [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). It will make your component a lot cleaner.